### PR TITLE
webdb-iv-challenge 

### PR DESCRIPTION
**Pull request by fncischen**

### CODE REVIEW

**Comment by fncischen** [comment]
The join method - for joining the associated recipes with the respective dish - is one of the highlights of today's lesson.

File: data/dishesDataModel/dishesDataModel.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    // .select("dishes.id", id)
+
+}
+
+function goShopping(dish, recipe) {
+    return db("dishes")
+    .where("dishes.name", dish)
+    .join("recipes", "recipes.dish_name", "=", "dishes.name")
+    .where("recipe_name", recipe)
+    .join("ingredients", "ingredients.recipe_name", "=", "recipe.recipe.name")
```
Comment (secantsquared) [inline data/dishesDataModel/dishesDataModel.js:31]: Good work on the joins.

File: data/recipiesDataModel/recipiesDataModel.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    return db("recipes").insert(recipe);
+}
+
+function getRecipesByDishName(dish_name) {
+    return db("recipes").where({"dish_name": dish_name});
+}
+
+function saveInstructions(ingredient) {
+    return db("ingredients").insert(ingredient).select("amount_of_ingredient", "units"); 
+}
```
Comment (secantsquared) [inline data/recipiesDataModel/recipiesDataModel.js:23]: Nice work on these functions. 
Well done.

File: migrations/20190328091027_create_recipe_table.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    return knex.schema.createTable('recipes', function(tbl){
+      tbl.increments();
+
+      tbl.string("recipe_name", 50)
+      .notNullable()
+      .unique();
+
+      tbl.string("dish_name").notNullable().unique();
+
+      tbl.foriegn("dish_name").references("name").inTable("dishes");
```
Comment (secantsquared) [inline migrations/20190328091027_create_recipe_table.js:12]:
```suggestion
      tbl.foreign("dish_name").references("name").inTable("dishes");
     
```

File: routers/dishesRouter.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+router.get("/:id/shoppinglist", (req,res) => {
+    if(!req.body.dish || ! req.body.recipe) {
+        return res.status(400).json({errorMessage: "You are missing a recipe or a dish for your shopping trip"})
+    }
+    else{
+        dishes.goShopping(req.body.dish, req.body.recipe)
+        .then(result => result.status(200).json(result))
+        .catch(() => res.status(500).json({errorMessage: "We had issues creating our traffic trip"}))
+    }
```
Comment (secantsquared) [inline routers/dishesRouter.js:32]: Great routes here, may want to consider some edge cases that may be needed, but I think you were still working on things - good work nonetheless.

**Review by secantsquared — APPROVED** [review]
💯

**Pull request merged**